paddle/fluid/framework/details/nccl_op_handle.h

```diff
@@ -0,0 +1,230 @@
+//   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "paddle/fluid/framework/details/op_handle_base.h"
+#include "paddle/fluid/framework/lod_tensor.h"
+#include "paddle/fluid/framework/scope.h"
+#include "paddle/fluid/platform/dynload/nccl.h"
+#include "paddle/fluid/platform/nccl_helper.h"
+
+DECLARE_bool(sync_nccl_allreduce);
+
+namespace paddle {
+namespace framework {
+namespace details {
+
+class NCCLOpHandleBase : public OpHandleBase {
+ public:
+  NCCLOpHandleBase(ir::Node* node, const std::vector<platform::Place>& places,
+                   const platform::MultiNCCLContextMap* nccl_ctxs)
+      : OpHandleBase(node), places_(places), nccl_ctxs_(nccl_ctxs) {
+    if (nccl_ctxs == nullptr) {
+      return;
+    }
+    // init device context
+    auto default_nccl_ctxs = nccl_ctxs_->DefaultFlatCtx();
+    for (auto& p : places_) {
+      this->SetDeviceContext(p, default_nccl_ctxs->DevCtx(p));
+    }
+  }
+  virtual ~NCCLOpHandleBase() {
+    for (auto& ev : inter_events_) {
+      PADDLE_ENFORCE(cudaEventDestroy(ev.second));
+    }
+    for (auto& ev : exter_events_) {
+      PADDLE_ENFORCE(cudaEventDestroy(ev.second));
+    }
+  }
+  void SetRunEnv(int run_order, bool use_hierarchical_allreduce) {
+    PADDLE_ENFORCE(run_order_ >= 0, "run_order must > 0");
```

Comment:
The condition is `run_order_ >= 0` but the info is `run_order must > 0`.

Reply:
Done.Thanks!